pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -69,20 +98,41 @@ func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTime
 				}
 			}
 		}()
+
+		go func() {
+			for {
+				select {
+				case <-p.close:
+					return
+				// Persist lazy loaded blocks in file every one minute
+				case <-time.After(1 * time.Minute):
```

Comment:
nit and maybe bikeshedding, but can this be part of the switch above? We can have only 1 goroutine per tenant as opposed to 2. And these are uncommon and quick events, so it's not likely they conflict

Reply:
If more than `select` case is using `time.After` and one of it is longer than the other, the longer `time.After` will never be executed because the `time.After` it is evaluated per iteration. That's why I am using two goroutines.

Reply:
oh i didn't realize we're using time.after and not time.ticker. I'm tempted to just move to time.ticker because it will make this function a bit shorter and will run less goroutines. But it's somewhat outside of the scope of this PR. If you think it's outside of scope, I can also open a PR to change this